yare-engine/src/test/java/com/sabre/oss/yare/invoker/java/MethodCallMetadataTest.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * MIT License
+ *
+ * Copyright 2018 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.yare.invoker.java;
+
+import com.sabre.oss.yare.engine.Collector;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static com.sabre.oss.yare.invoker.java.MethodCallMetadata.method;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class MethodCallMetadataTest {
```

Comment:
Shouldn't we split that test into separate tests for MethodCallMetadataValidator and MethodCallMetadata?

Reply:
What we want to test in fact is the "method(" call in MethodCallMetadataFactory to work correctly for various 'method mappings' - fail fast on engine creation rather than in runtime. Testing MethodCallMetadataValidator separate from MethodCallMetadataFactory doesn't make much sense to me.

Reply:
As I can see the Validator is package private, which confirms that validator as factory's integral part (as mentioned in the comment above) I am fine with that.